llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -2039,6 +2039,8 @@ bool VectorCombine::foldShuffleToIdentity(Instruction &I) {
         continue;
       } else if (auto *II = dyn_cast<IntrinsicInst>(FrontU);
                  II && isTriviallyVectorizable(II->getIntrinsicID())) {
+        if (II->hasOperandBundles())
```

Comment:
can we not put this in the if condition above?

Reply:
I've updated to check both the first lane and the others, in case the intrinsic is different.